### PR TITLE
ActionsCheckbox, Validation, Loading Indicator, etc..

### PR DESCRIPTION
**Pull request by jdtoombs**
Closing my old PR and opening a new one. Just didn't want to risk anything...

This PR includes:

- ActionCheckbox component to handle the logic of sending the area of actions to the API based on user input
- Validation for ContentForm
- Loading indicators for Data Sources, and Content List

### CODE REVIEW

File: app/editor/src/components/form/datepicker/SelectDate.tsx [hunk 1]
```diff
@@ -53,7 +53,11 @@ export const SelectDate: React.FC<IDatePickerProps> = ({
 
   return (
     <styled.SelectDate className="frm-in" variant={variant} required={required} width={width}>
-      {label && <label htmlFor={id ?? `txt-${name}`}>{label}</label>}
+      {label && (
+        <label className={required ? 'required' : ''} htmlFor={id ?? `txt-${name}`}>
```
Comment (Fosol) [inline app/editor/src/components/form/datepicker/SelectDate.tsx:57]: You can do this without the extra className.  I believe the parent object will have the attribute `required`, it'll also be available in the styled component.

File: app/editor/src/components/form/text/FormSizing.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export enum FormInputSizes {
```
Comment (Fosol) [inline app/editor/src/components/form/text/FormSizing.ts:1]: I thought we were using the `FieldSize`?

File: app/editor/src/components/formik/datepicker/FormikDatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        tooltip={tooltip}
+        children={children}
+        className={className}
+        selectedDate={selectedDate}
+        required={required}
+        width={width}
+        onChange={onChange}
+        {...rest}
+      />
+      {error ? <p role="alert">{error}</p> : null}
```
Comment (Fosol) [inline app/editor/src/components/formik/datepicker/FormikDatePicker.tsx:43]: I think I've moved all the error logic to the base component.  See other form examples.

File: app/editor/src/components/formik/select/FormikSelect.tsx [hunk 4]
```diff
@@ -40,6 +39,7 @@ export const FormikSelect = <OptionType extends IOptionItem>({
         error={error}
         {...rest}
       />
+      <p role="alert">{error}</p>
```
Comment (Fosol) [inline app/editor/src/components/formik/select/FormikSelect.tsx:42]: This was moved to the base component.

File: app/editor/src/components/formik/text/FormikText.tsx [hunk 5]
```diff
@@ -33,6 +33,7 @@ export const FormikText = <T,>({
         variant={disabled ? TextVariant.disabled : TextVariant.primary}
         {...rest}
       ></Text>
+      {error ? <p role="alert">{error}</p> : null}
```
Comment (Fosol) [inline app/editor/src/components/formik/text/FormikText.tsx:36]: This was moved to the base component.

File: libs/npm/core/package.json [hunk 6]
```diff
@@ -1,113 +1,113 @@
...
+        "src/**/*",
+        "package.json"
     ],
-    "plugins": [
-      "simple-import-sort"
-    ],
-    "rules": {
-      "simple-import-sort/imports": "error",
-      "simple-import-sort/exports": "error"
+    "main": "dist/index.js",
```
Comment (Fosol) [inline libs/npm/core/package.json:12]: This makes me nervous, what changed?

File: app/editor/src/components/form/text/FormSizing.ts (hunk 2 above)
Comment (jdtoombs) [reply app/editor/src/components/form/text/FormSizing.ts:1]: Right we are! Need to remove this. Snuck in on my other branch

File: libs/npm/core/package.json (hunk 6 above)
Comment (jdtoombs) [reply libs/npm/core/package.json:12]: Let me get rid of that..

File: app/editor/src/hooks/api-editor/constants/ActionName.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export enum ActionName {
```
[the remark this answers is not in the document]
Comment (jdtoombs) [reply app/editor/src/hooks/api-editor/constants/ActionName.ts:1]: Was for the ActionCheckbox for when your passing it the props. Not required but
eg)
 
image

File: app/editor/src/components/form/datepicker/SelectDate.tsx (hunk 1 above)
Comment (jdtoombs) [reply app/editor/src/components/form/datepicker/SelectDate.tsx:57]: Will take a look at that

**Review by sturufous — APPROVED** [review]

File: app/editor/src/components/formik/select/FormikSelect.tsx (hunk 4 above)
Comment (jdtoombs) [reply app/editor/src/components/formik/select/FormikSelect.tsx:42]: It's seems although the Formik error messages conflict with the logic implemented at the base level. Should we keep these errors separate? To me I think it makes sense to have the Formik/Yup errors logic and display controlled in this component. If not, I can try to rewrite the base level so it doesn't override

File: app/editor/src/components/form/datepicker/SelectDate.tsx (hunk 1 above)
Comment (jdtoombs) [reply app/editor/src/components/form/datepicker/SelectDate.tsx:57]: Looks like required attribute does not get added to div elements

File: app/editor/src/components/formik/select/FormikSelect.tsx (hunk 4 above)
Comment (Fosol) [reply app/editor/src/components/formik/select/FormikSelect.tsx:42]: the display of the error should propagate to the base component.  All it should take is setting the base component with the error `error={error}` I think.

File: libs/npm/core/package.json [hunk 8]
```diff
@@ -1,113 +1,113 @@
 {
-  "name": "tno-core",
```
Comment (jdtoombs) [inline libs/npm/core/package.json:2]: My linter makes this look scarier than it is

File: app/editor/src/features/content/form/PropertiesContentForm.tsx [hunk 9]
```diff
@@ -1,137 +1,253 @@
...
                 value={values.publishedOn}
-                disabled
                 onChange={(date: any) => {
-                  setContent({ ...content, publishedOn: date });
+                  setFieldValue('publishedOn', date);
                 }}
               />
             </Col>
-            <Col style={{ marginTop: '0.45em' }}>
+            <Col style={{ marginLeft: '1%' }}>
```
Comment (Fosol) [inline app/editor/src/features/content/form/PropertiesContentForm.tsx:142]: Need to get in a habit of moving styling to the styled-component.

**Review by Fosol — APPROVED** [review]

File: app/editor/src/features/content/form/PropertiesContentForm.tsx (hunk 9 above)
Comment (jdtoombs) [reply app/editor/src/features/content/form/PropertiesContentForm.tsx:141]: Yeah there's a bunch I need to move. Will create a tech debt ticket for myself. Also ready to merge!

**Review by Fosol — APPROVED** [review]

**Pull request merged**